### PR TITLE
Attachments update

### PR DESCRIPTION
**Pull request by hhdivil**
### Is the pull request based off the latest version?

Yes

### What features have you added?

Mainly IncomingMessageAttachment class which wraps up some of the messy array stuff, not sure what other getters to add though, any thoughts as to what other people commonly access?  See commits for the rest.

### What bugs did you fix?

N/A

### Is your code valid PSR-2?

Believe so

### Have you properly documented your code?

Yes

### Has anything in your pull request already been fixed?

N/A

### Anything else?

Added some stuff to the docs for this one as it changes some of the way you access attachments which may break existing scripts using this library.  What are your thoughts on that?

### Optional things below
[] Added your name to the credits
[] Add a update name below:

### CODE REVIEW

**Comment by hhdivil** [comment]
yeah that's a good call i'll sort that

**Comment by xXXIMMATTHEWXXx** [comment]
Be sure to add that before I merge

File: ImapClient/IncomingMessageAttachment.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace SSilence\ImapClient;
+
```
Comment (xXXIMMATTHEWXXx) [inline ImapClient/IncomingMessageAttachment.php]: Copy right and class info needed

File: ImapClient/IncomingMessageAttachment.php [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+class IncomingMessageAttachment {
+
+	private $rawObject;
+
+	public function __construct ($rawObject) {
```
Comment (xXXIMMATTHEWXXx) [inline ImapClient/IncomingMessageAttachment.php]: Function doc

File: ImapClient/IncomingMessageAttachment.php [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+		$this->rawObject = $rawObject;
+	}
+
```
Comment (xXXIMMATTHEWXXx) [inline ImapClient/IncomingMessageAttachment.php]: Function doc

File: ImapClient/IncomingMessageAttachment.php [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+	public function getRaw () {
+		return $this->rawObject;
+	}
+
+	private $name;
```
Comment (xXXIMMATTHEWXXx) [inline ImapClient/IncomingMessageAttachment.php]: Function doc. Vars should be at the top of a class

File: ImapClient/IncomingMessageAttachment.php [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+				if ($param->attribute == 'filename') {
+					$this->name = $param->value;
+					break;
+				};
+			};
+		};
+		return $this->name;
+	}
+
+	private $body;
```
Comment (xXXIMMATTHEWXXx) [inline ImapClient/IncomingMessageAttachment.php]: No vars before methods. Function docs

File: docs/guide-en/IncomingMessage.md [hunk 6]
```diff
@@ -168,8 +168,30 @@ $imap->incomingMessage->message->html and $imap->incomingMessage->message->plain
 Are auto generated by the client. This allows you to have an html and a plain text version of the email
 
 ### Attachment properties
```
Comment (xXXIMMATTHEWXXx) [inline docs/guide-en/IncomingMessage.md]: Please added the new methods to Methods.md

File: ImapClient/IncomingMessage.php [hunk 7]
```diff
@@ -174,38 +174,70 @@ private function getRecursiveSections($obj, $recursive = 1)
         };
     }
 
+	 /**
+	  * Gets all sections, or if parameter is specified sections by type
+	  * @param string $type
+	  * @return array
+	  */
+	 private function getSections ($type = null) {
```
Comment (xXXIMMATTHEWXXx) [inline ImapClient/IncomingMessage.php]: Add to Methods.md

**Comment by hhdivil** [comment]
Will open another fresh one for this

**Pull request closed without merging**